scripts/synthetix/facilitate_trade.js

```diff
@@ -73,44 +88,89 @@ module.exports = async (callback) => {
     const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
 
-    // Compute buy-sell amounts based on unlimited orders with rates from above.
-    const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
-      1 / (formatedRate * (1 - sUSDTosETHFee)),
-      sETH.decimals,
-      sUSD.decimals
-    )
-    const { base: sellETHAmount, quote: buySUSDAmount } = getUnlimitedOrderAmounts(
-      formatedRate * (1 + sETHTosUSDFee),
-      sUSD.decimals,
-      sETH.decimals
-    )
-
-    const buyAmounts = [buyETHAmount, buySUSDAmount]
-    const sellAmounts = [sellSUSDAmount, sellETHAmount]
-
-    // Fetch auction index and declare validity interval for orders.
-    // Note that order validity interval is inclusive on both sides.
-    const batchId = (await exchange.getCurrentBatchId.call()).toNumber()
-    const validFroms = Array(2).fill(batchId)
-    const validTos = Array(2).fill(batchId)
-
-    // Avoid querying exchange by tokenAddress for fixed tokenId
-    const buyTokens = [sETH, sUSD].map((token) => token.exchangeId)
-    const sellTokens = [sUSD, sETH].map((token) => token.exchangeId)
-
-    const gasPrices = await (await fetch(gasStationURL[networkId])).json()
-    const scaledGasPrice = parseInt(gasPrices[argv.gasPrice] * argv.gasPriceScale)
-    console.log(`Using current "${argv.gasPrice}" gas price scaled by ${argv.gasPriceScale}: ${scaledGasPrice}`)
-    await exchange.placeValidFromOrders(buyTokens, sellTokens, validFroms, validTos, buyAmounts, sellAmounts, {
-      from: account,
-      gasPrice: scaledGasPrice,
-    })
+    // Initialize order array.
+    const orders = []
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above when the price is right.
+    const ourBuyETHRate = formatedRate * (1 - sUSDTosETHFee)
+    if (ourBuyETHRate > 1 / exchangeBuyETHPrice) {
+      // If we are willing to pay more than the exchange quotes as buy price.
+      console.log(`Placing an order to buy sETH at ${ourBuyETHRate}`)
+      const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
+        1 / ourBuyETHRate,
+        sETH.decimals,
+        sUSD.decimals
+      )
+      orders.push({
+        buyToken: sETH.exchangeId,
+        sellToken: sUSD.exchangeId,
+        buyAmount: buyETHAmount,
+        sellAmount: sellSUSDAmount,
+      })
+    } else {
+      console.log(`Not placing buy  sETH order, our rate of ${ourBuyETHRate.toFixed(2)} is too low  for exchange.`)
+    }
+
+    const ourSellETHRate = formatedRate * (1 + sETHTosUSDFee)
+    if (ourSellETHRate < exchangeSellETHPrice) {
+      // If we are willing to sell at a price less than exchange quote.
+      console.log(`Placing an order to sell sETH at ${ourSellETHRate}`)
+      const { base: sellETHAmount, quote: buySUSDAmount } = getUnlimitedOrderAmounts(
+        formatedRate * (1 + sETHTosUSDFee),
```

Comment:
Very nitpick, but why not using `ourSellETHRate` here, similar to the previous call to `getUnlimitedOrderAmounts`?
```suggestion
        ourSellETHRate,
```